network/netconf/config.go

```diff
@@ -114,4 +114,19 @@ type AlspConfig struct {
 	// HeartBeatInterval is the interval between heartbeats sent by the ALSP module. The heartbeats are recurring
 	// events that are used to perform critical ALSP tasks, such as updating the spam records cache.
 	HearBeatInterval time.Duration `mapstructure:"alsp-heart-beat-interval"`
+
+	// SyncEngineBatchRequestBaseProb is the base probability in [0,1] that's used in creating the final probability of creating a
+	// misbehavior report for a BatchRequest message. This is why the word "base" is used in the name of this field,
+	// since it's not the final probability and there are other factors that determine the final probability.
+	// The reason for this is that we want to increase the probability of creating a misbehavior report for a large batch.
+	SyncEngineBatchRequestBaseProb float32 `mapstructure:"alsp-sync-engine-batch-request-base-prob"`
+
+	// SyncEngineRangeRequestBaseProb is the base probability in [0,1] that's used in creating the final probability of creating a
+	// misbehavior report for a RangeRequest message. This is why the word "base" is used in the name of this field,
+	// since it's not the final probability and there are other factors that determine the final probability.
+	// The reason for this is that we want to increase the probability of creating a misbehavior report for a large range.
+	SyncEngineRangeRequestBaseProb float32 `mapstructure:"alsp-sync-engine-range-request-base-prob"`
+
+	// SyncEngineSyncRequestProb is the probability in [0,1] of creating a misbehavior report for a SyncRequest message.
+	SyncEngineSyncRequestProb float32 `mapstructure:"alsp-sync-engine-sync-request-prob"`
```

Comment:
I suggest adding validation tags to make sure these are greater than 0
```suggestion
	SyncEngineBatchRequestBaseProb float32 `validate:"gt=0" mapstructure:"alsp-sync-engine-batch-request-base-prob"`

	// SyncEngineRangeRequestBaseProb is the base probability in [0,1] that's used in creating the final probability of creating a
	// misbehavior report for a RangeRequest message. This is why the word "base" is used in the name of this field,
	// since it's not the final probability and there are other factors that determine the final probability.
	// The reason for this is that we want to increase the probability of creating a misbehavior report for a large range.
	SyncEngineRangeRequestBaseProb float32 `validate:"gt=0" mapstructure:"alsp-sync-engine-range-request-base-prob"`

	// SyncEngineSyncRequestProb is the probability in [0,1] of creating a misbehavior report for a SyncRequest message.
	SyncEngineSyncRequestProb float32 `validate:"gt=0" mapstructure:"alsp-sync-engine-sync-request-prob"`
```

Reply:
Great idea. I made the validation enforce a range of values between 0 and 1.

26e817fdf608eb4bfe4151638c78c44e8c07dd95

Reply:
Also, please consider encapsulating them into a sync engine-specific config type: 

```go
// AlspConfig is the config for the Application Layer Spam Prevention (ALSP) protocol.
type AlspConfig struct {
        //... Other Configs

	SyncEngine SyncEngineAlspConfig
}

type SyncEngineAlspConfig struct {
	// SyncEngineBatchRequestBaseProb is the base probability in [0,1] that's used in creating the final probability of creating a
	// misbehavior report for a BatchRequest message. This is why the word "base" is used in the name of this field,
	// since it's not the final probability and there are other factors that determine the final probability.
	// The reason for this is that we want to increase the probability of creating a misbehavior report for a large batch.
	BatchRequestBaseProb float32 `validate:"range=0,1" mapstructure:"alsp-sync-engine-batch-request-base-prob"`

	// SyncEngineRangeRequestBaseProb is the base probability in [0,1] that's used in creating the final probability of creating a
	// misbehavior report for a RangeRequest message. This is why the word "base" is used in the name of this field,
	// since it's not the final probability and there are other factors that determine the final probability.
	// The reason for this is that we want to increase the probability of creating a misbehavior report for a large range.
	RangeRequestBaseProb float32 `validate:"range=0,1" mapstructure:"alsp-sync-engine-range-request-base-prob"`

	// SyncEngineSyncRequestProb is the probability in [0,1] of creating a misbehavior report for a SyncRequest message.
	SyncRequestProb float32 `validate:"range=0,1" mapstructure:"alsp-sync-engine-sync-request-prob"`
}
```

Reply:
Great idea. Now the SyncEngine ALSP config is referenced list this:

```
		batchRequestBaseProb: flowConfig.NetworkConfig.SyncEngine.SyncEngineBatchRequestBaseProb,
		syncRequestProb:      flowConfig.NetworkConfig.SyncEngine.SyncEngineSyncRequestProb,
		rangeRequestBaseProb: flowConfig.NetworkConfig.SyncEngine.SyncEngineRangeRequestBaseProb,
```

5a5439d0eef5cc091b5536935348232740273a4c

Reply:
Shortened field names to this:

```
batchRequestBaseProb: flowConfig.NetworkConfig.SyncEngine.BatchRequestBaseProb,
syncRequestProb:      flowConfig.NetworkConfig.SyncEngine.SyncRequestProb,
rangeRequestBaseProb: flowConfig.NetworkConfig.SyncEngine.RangeRequestBaseProb,
```

590e81b796562340d80f56738a0ea9cd690107c0